src/signal_scope/signalScopeSetup.py

```diff
@@ -203,18 +203,45 @@ def _signalFunction(x):
     _mainWindow.addPythonSignal(plot, [channel, _signalFunction, label or signalFunction.__doc__, color])
 
 
+def addSignalFunctions(channel, signalFunction, keys, keyLookup=None, plot=None, colors=None, labels=None):
+    
+    def func(key, keyStr):
+        f = lambda msg: (signalFunction(msg)[0], signalFunction(msg)[1][key])
```

Comment:
May be instead of a lambda you could define a local function, then you wouldn't have to call signalFunction twice.

    def f(msg):
        time, values = signalFunction(msg)
        return time, values[key]